packages/ckeditor5-dev-release-tools/lib/tasks/cleanuppackages.js

```diff
@@ -0,0 +1,175 @@
+		packageJsonFieldsToRemove,
+		cwd: upath.normalizeTrim( cwd )
+	};
+}
+
+/**
+ * Remove unnecessary files and directories from the package directory.
+ *
+ * @param {Object} packageJson
+ * @param {String} packagePath
+ */
+function cleanUpPackageDirectory( packageJson, packagePath ) {
+	if ( packageJson.files ) {
+		// Find and remove files that don't match the `files` field in the `package.json`.
+		const files = globSync( '**', {
+			cwd: packagePath,
+			absolute: true,
+			nodir: true,
+			ignore: [
+				'README.md',
+				'LICENSE.md',
+				'package.json',
+				...getIgnoredFilePatterns( packageJson )
+			]
+		} );
+
+		for ( const file of files ) {
+			fs.removeSync( file );
+		}
+	}
+
+	// Find and remove empty directories in the package directory.
+	const directories = globSync( '**/', {
+		cwd: packagePath,
+		absolute: true
+	} )
+		.map( path => upath.normalize( path ) )
+		.sort( sortPathsFromDeepestFirst );
+
+	for ( const directory of directories ) {
+		const isEmpty = fs.readdirSync( directory ).length === 0;
+
+		if ( isEmpty ) {
+			fs.removeSync( directory );
+		}
+	}
+
+	// Remove `node_modules`.
+	fs.removeSync( upath.join( packagePath, 'node_modules' ) );
+}
+
+/**
+ * Creates an array of patterns to ignore for the `glob` calls.
+ *
+ * @param {Object} packageJson
+ * @returns {Array.<String>}
+ */
+function getIgnoredFilePatterns( packageJson ) {
+	// The patterns supported by `package.json` in the `files` field do not correspond 1:1 to the patterns expected by the `glob`.
+	// For this reason, we always treat each pattern from `files` as if it was the beginning of a path to match - not just a final path.
+	//
+	// Example: for the entry `src` we prepare the `src/**` pattern for `glob`.
+	//
+	// If the globstar pattern (`**`) is alone in a path portion, then it matches zero or more directories and subdirectories.
+	const patterns = packageJson.files.map( pattern => pattern + '/**' );
+
+	if ( packageJson.main ) {
+		patterns.push( packageJson.main );
+	}
+
+	return patterns;
+}
+
+/**
+ * Removes unnecessary fields from the `package.json`.
+ *
+ * @param {Object} packageJson
+ * @param {Array.<String>} packageJsonFieldsToRemove
+ */
+function cleanUpPackageJson( packageJson, packageJsonFieldsToRemove ) {
+	for ( const key of Object.keys( packageJson ) ) {
+		if ( packageJsonFieldsToRemove.includes( key ) ) {
+			delete packageJson[ key ];
+		}
+	}
+}
+
+/**
+ * Sort function that defines the order of the paths. It sorts paths from the most nested ones first.
+ *
+ * @param {String} firstPath
+ * @param {String} secondPath
+ * @returns {Number}
+ */
+function sortPathsFromDeepestFirst( firstPath, secondPath ) {
+	const firstPathSegments = firstPath.split( '/' ).length;
+	const secondPathSegments = secondPath.split( '/' ).length;
+
+	if ( firstPathSegments > secondPathSegments ) {
+		return -1;
+	}
+
+	if ( firstPathSegments < secondPathSegments ) {
+		return 1;
+	}
+
+	return 0;
```

Comment:
This can be simpler:

```suggestion
	return secondPathSegments - firstPathSegments;
```

Also, removing the sorting function alltogether does not fail any unit tests, so a test covering the use case should be added.

Reply:
You are right, thanks for finding this. It turns out that in the tests I only checked the existence of the required files or directories. I did not verify the absence of unnecessary files or directories.